snuba/cli/query_comparer.py

```diff
@@ -139,86 +53,186 @@ def query_comparer(
+
+            assert (
+                v1_result.query_id == v2_result.query_id
+            ), f"Invalid query_ids: {v1_result.query_id} and {v2_result.query_id} must match"
+            total_rows += 1
+            if v1_result == v2_result:
+                continue
+
+            assert isinstance(v1_result, QueryMeasurementResult)
+            assert isinstance(v2_result, QueryMeasurementResult)
+
+            perf_mismatch = create_perf_mismatch(v1_result, v2_result)
+            data_mismatch = create_data_mismatch(v1_result, v2_result)
+            if perf_mismatch:
+                perf_mismatches.append(perf_mismatch)
+            if data_mismatch:
+                data_mismatches.append(data_mismatch)
+
+        def save_comparisons(compare_type: str, mismatches: MismatchResultSet) -> None:
+            compared_file_format = FileFormat(
+                f"compared-{compare_type}",
+                result_file_format.date,
+                result_file_format.table,
+                result_file_format.hour,
+            )
+            file_manager.save(compared_file_format, mismatches, header_row=True)
+            # need the filename for the slack report
+            filename = file_manager.format_filename(compared_file_format)
+            _send_slack_report(
+                compare_type, total_rows, mismatches, result_file_format.table, filename
+            )
+
+        if compare_type == "perf":
+            save_comparisons(compare_type, perf_mismatches)
+        elif compare_type == "data":
+            save_comparisons(compare_type, data_mismatches)
+        else:
+            # no compare type == run both reports
+            save_comparisons("perf", perf_mismatches)
+            save_comparisons("data", data_mismatches)
+
+
+PERF_THRESHOLDS = {
+    "query_duration_ms": 0,
+    "read_rows": 0,
+    "read_bytes": 0,
+}
+
+DATA_THRESHOLDS = {
+    "result_rows": 0,
+    "result_bytes": 0,
+}
+
+
+def create_perf_mismatch(
+    v1_data: QueryMeasurementResult, v2_data: QueryMeasurementResult
+) -> Optional[PerfMismatchResult]:
+    d_ms_1 = v1_data.query_duration_ms
+    d_ms_2 = v2_data.query_duration_ms
+    d_ms_delta = d_ms_2 - d_ms_1
+
+    rd_r_1 = v1_data.read_rows
+    rd_r_2 = v2_data.read_rows
+    rd_r_delta = rd_r_2 - rd_r_1
+
+    rd_b_1 = v1_data.read_bytes
+    rd_b_2 = v2_data.read_bytes
+    rd_b_delta = rd_b_2 - rd_b_1
+
+    if (
+        (d_ms_delta > PERF_THRESHOLDS["query_duration_ms"])
+        or rd_r_delta > PERF_THRESHOLDS["read_rows"]
+        or rd_b_delta > PERF_THRESHOLDS["read_bytes"]
+    ):
+        # mismatch!
+        return PerfMismatchResult(
+            v1_data.query_id,
+            d_ms_1=d_ms_1,
+            d_ms_2=d_ms_2,
+            rd_r_1=rd_r_1,
+            rd_r_2=rd_r_2,
+            rd_b_1=rd_b_1,
+            rd_b_2=rd_b_2,
+            d_ms_delta=d_ms_delta,
+            rd_r_delta=rd_r_delta,
+            rd_b_delta=rd_b_delta,
+        )
```

Comment:
Let's rename the fields to be more explicit so that the CSV headers are easier to read when analyzing them.

Reply:
@nikhars would you prefer to have the full names written out? `duration_ms_1`, `read_rows_1`, etc

Reply:
yeah that should work. what does `_1` and `_2` signify. Maybe expand on that as well